schema_validation/validate_schema.go

```diff
@@ -47,24 +47,36 @@ type SchemaValidator interface {
 	ValidateSchemaBytes(schema *base.Schema, payload []byte) (bool, []*liberrors.ValidationError)
 }
 
+// Config  Holds Schema Validation options
+type Config struct {
+	RegexEngine jsonschema.RegexpEngine // Use the given Regular Expression Engine
+}
+
 var instanceLocationRegex = regexp.MustCompile(`^/(\d+)`)
 
 type schemaValidator struct {
+	Config
 	logger *slog.Logger
 	lock   sync.Mutex
 }
 
 // NewSchemaValidatorWithLogger will create a new SchemaValidator instance, ready to accept schemas and payloads to validate.
-func NewSchemaValidatorWithLogger(logger *slog.Logger) SchemaValidator {
-	return &schemaValidator{logger: logger, lock: sync.Mutex{}}
+func NewSchemaValidatorWithLogger(logger *slog.Logger, config ...Config) SchemaValidator {
+
+	cfg := Config{} // Default Config
+	if len(config) > 0 {
+		cfg = config[0]
+	}
```

Comment:
Why using Config here and not Option?

Reply:
This might be one that I failed to remediate... will check.